portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,9 +39,23 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Task> tasks = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
A note about Java: Casting can be somewhat risky. In this case, [getProperty()](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#getproperty) is only guaranteed to return an Object. In this specific case, you know that the "comment" property will be an Object that is a String but you can imagine in a more complex application you might not be able to guarantee that. Maybe it would return a more general Object that can't be cast to a String.

One thing you can do to be sure it's safe to cast is to use the instanceof operator. [This](https://stackoverflow.com/questions/7313559/what-is-the-instanceof-operator-used-for-in-java) is a good explanation of that.

I know there is some controversy over the use of instanceof, but I'm not aware of a better way to handle that situation. Also, I think your code should be fine here, I just wanted to make you aware of this (and you can add instanceof if you want to practice it).